refiner-lib/src/refiner_inner.rs

```diff
@@ -631,6 +677,35 @@ fn fill_tx(
         .s(U256::zero())
 }
 
+/// Extracts an address from a Deploy transaction execution status
+fn extract_address_deploy(status: &ExecutionStatusView) -> Result<Address, RefinerError> {
+    if let ExecutionStatusView::SuccessValue(value_str) = status {
+        let value_vec = base64::decode(value_str).map_err(RefinerError::SuccessValueBase64Args)?;
+
+        let submit_result =
+            SubmitResult::try_from_slice(&value_vec).map_err(|_| RefinerError::FailNearTx)?;
+
+        if let aurora_engine::parameters::TransactionStatus::Succeed(address_vec) =
+            submit_result.status
+        {
+            return Address::try_from_slice(&address_vec).map_err(|_| RefinerError::FailNearTx);
+        }
+    }
+
+    Err(RefinerError::FailNearTx)
+}
+
+/// Extracts an address from a DeployErc20 transaction execution status
+fn extract_address_deploy_erc20(status: &ExecutionStatusView) -> Result<Address, RefinerError> {
+    if let ExecutionStatusView::SuccessValue(value_str) = status {
+        let address_vec =
+            base64::decode(value_str).map_err(RefinerError::SuccessValueBase64Args)?;
+        Address::try_from_slice(&address_vec).map_err(|_| RefinerError::FailNearTx)
```

Comment:
Is there a better error we can use here? It's not that the transaction failed, if we fail to parse then something else is wrong, for example maybe the Refiner code is out out date compared with the deployed Engine contract.